cpp/core/memory/ColumnarBatch.h

```diff
@@ -47,7 +48,13 @@ class ColumnarBatch {
   virtual std::shared_ptr<ArrowSchema> exportArrowSchema() = 0;
 
   virtual int64_t getExportNanos() const;
-  ;
+
+  virtual int32_t getRowSize() const;
+
+  virtual std::shared_ptr<ColumnarBatch> getNonPartitionedColumns(
+      std::vector<size_t> partitionColIndiceVec,
+      MemoryManager* memoryManager,
+      char*& rowBytes) const;
```

Comment:
What's the purpose of this function?

Reply:
@zhztheplayer Partition write only need to write the non partition columns into the file. This method is to retrieve the non partition columns  based on the partitionColIndiceVec.

Reply:
We did add a `::select` API to ExecutionCtx. Is that one suitable for this work?

Reply:
@zhztheplayer It appears that the ::select API can be useful for my purpose in ExecutionCtx. Spark needs to create the partition folder based on the first row in the column batch. Therefore, we only need to return the bytes of the first row to Spark, rather than the entire columnar batch. How about adding a getRowBytes method to ColumnarBatch specifically designed to retrieve the bytes of the first row?

Reply:
If the output `getRowBytes` is compatible with Spark's UnsafeRow / InternalRow then it's OK to add it to `ColumnarBatch`.

The function should take row ordinal as its parameter, say `getRowBytes(int32_t rowId)` or something like that.

Reply:
@zhztheplayer Yes. The getRowBytes method is compatible with Spark's UnsafeRow / InternalRow. I have made improvements by refining the method to getFirstRowBytes, which now only returns the bytes of the first row. Could you please review it again ? Thank you.